pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -99,35 +99,48 @@ func UpdateContainersEntrypointsIfNeeded(
 	devfileRunCmd string,
 	devfileDebugCmd string,
 ) ([]corev1.Container, error) {
-	buildCmd, err := libdevfile.GetBuildCommand(devfileObj.Data, devfileBuildCmd)
+	buildCommand, hasBuildCmd, err := libdevfile.GetCommand(devfileObj, devfileBuildCmd, v1alpha2.BuildCommandGroupKind)
 	if err != nil {
 		return nil, err
 	}
-	runCommand, err := libdevfile.GetRunCommand(devfileObj.Data, devfileRunCmd)
+	runCommand, hasRunCmd, err := libdevfile.GetCommand(devfileObj, devfileRunCmd, v1alpha2.RunCommandGroupKind)
 	if err != nil {
 		return nil, err
 	}
-	debugCommand, err := libdevfile.GetDebugCommand(devfileObj.Data, devfileDebugCmd)
+	debugCommand, hasDebugCmd, err := libdevfile.GetCommand(devfileObj, devfileDebugCmd, v1alpha2.DebugCommandGroupKind)
 	if err != nil {
 		return nil, err
 	}
 
+	var cmdsToHandle []v1alpha2.Command
+	if hasBuildCmd {
+		cmdsToHandle = append(cmdsToHandle, buildCommand)
+	}
+	if hasRunCmd {
+		cmdsToHandle = append(cmdsToHandle, runCommand)
+	}
+	if hasDebugCmd {
+		cmdsToHandle = append(cmdsToHandle, debugCommand)
+	}
+
 	var components []string
 	var containerComps []string
-	for _, cmd := range []v1alpha2.Command{buildCmd, runCommand, debugCommand} {
+	for _, cmd := range cmdsToHandle {
 		containerComps, err = libdevfile.GetContainerComponentsForCommand(devfileObj, cmd)
 		if err != nil {
 			return nil, err
 		}
 		components = append(components, containerComps...)
 	}
 
-	for i := range containers {
-		container := &containers[i]
-		for _, c := range components {
-			if container.Name == c {
-				overrideContainerCommandAndArgsIfNeeded(container)
-				break
+	if len(components) != 0 {
```

Comment:
You could start by looping into components first, so you won't need this if len(components)

Reply:
That was a trade-off I made so as not to uselessly iterate over any of those slices if any of them was empty.
But you're right - this might be premature optimization because they are not that big. I'll change this as suggested.

Reply:
ok, I didn't get you were checking the size of the two arrays this way. As you prefer